src/new/connection.ts

```diff
@@ -99,6 +113,89 @@ class WalletConnection extends BaseConnection {
       this.websocket.on('subscribe_address', handleSubscribeAddress());
     }
   }
+
+  streamEndHandler() {
+    this.streamController?.abort();
+    this.streamController = null;
+  }
+
+  lockStream(streamId: string): boolean {
+    if (this.streamController === null) {
+      this.streamController = new StreamController(streamId);
+      return true;
+    }
+    return false;
+  }
+
+  startStreamingHistory(id: string, firstIndex: number, xpubkey: string, gapLimit: number = -1) {
+    if (this.streamController?.streamId !== id) {
+      throw new Error('There is an on-going stream, cannot start a second one');
+    }
+    if (this.websocket) {
```

Comment:
Maybe raise an exception in the else block?

Reply:
Done

Reply:
Thanks!